### PR TITLE
Refactor request API implementation into common, grpc, and http

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

File: src/clients/c++/request_grpc.h [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+/// \file
```
Comment (GuanLuo) [inline src/clients/c++/request_grpc.h]: extra comment?

File: src/clients/c++/request_http.h [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+/// \file
```
Comment (GuanLuo) [inline src/clients/c++/request_http.h]: extra?

File: src/clients/c++/request_grpc.h (hunk 1 above)
Comment (deadeyegoodwin) [reply src/clients/c++/request_grpc.h]: Do you mean the \file? That's required to get API documentation generated automatically.

Comment (GuanLuo) [reply src/clients/c++/request_grpc.h]: oh I see

**Pull request merged**